### PR TITLE
AYS-614 | Password Length Validation

### PR DESCRIPTION
**Pull request by htczrrsz**
## Pull Request

**Description:**

- Added password length validation tests for minimum and maximum length passwords.

**Changes Made:**

- TokenTest Class: Added password length validation and edge case tests. Introduced a new invalidPasswordAndEdgeCases DataProvider to test various invalid password cases, such as blank, passwords less than 8 characters, and greater than 128 characters.

- AysErrorMessage: Added a new error message for password validation related to the length constraint.

- AysRandomUtil: Added a generatePasswordWithinRange method to help generate passwords of varying lengths for testing edge cases.

**Additional Information:**
Add any additional information that might be relevant to reviewers.

**Reviewer Instructions:**
Provide instructions or guidelines for reviewers to follow when assessing your pull request.

### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be-api-automation/blob/main/CONTRIBUTING.md#pull-request-naming-conventions)
      .
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be-api-automation/blob/main/CONTRIBUTING.md#branch-naming-conventions)
      .

- [x] **Local Testing:**
    - I have added positive and negative test scenarios (if applicable)
    - All tests are passing.
    - I tagged all test as smoke or regression
    - I added newly added tests to related xml file in test suite

- [x] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated.

- [x] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

- [x] **Reviewer Instructions:**
    - Provide instructions or guidelines for reviewers to follow when assessing your pull request.

### CODE REVIEW

File: src/test/java/org/ays/auth/tests/TokenTest.java [hunk 1]
```diff
@@ -42,19 +43,20 @@ public void getTokenWithUnAuthUserEmailAddress() {
...
+        loginPayload.setPassword(AysRandomUtil.generatePassword());
         Response response = AuthEndpoints.token(loginPayload);
         response.then()
                 .spec(AysResponseSpecs.expectUnauthorizedResponseSpec());
     }
 
-    @Test(groups = {"Regression"})
-    public void getTokenWithNullPassword() {
+    @Test(groups = {"Regression"}, dataProvider = "invalidPasswordAndEdgeCases", dataProviderClass = AysDataProvider.class)
+    public void getTokenWithInvalidPasswordAndEdgeCases(String password, AysErrorMessage errorMessage, String field, String type) {
```
Comment (Nbhtozcn) [inline src/test/java/org/ays/auth/tests/TokenTest.java:53]: Burdaki edge cases de invalid data oldugu icin getTokenWithInvalidPasswordData demek daha uygun olur.

File: src/test/java/org/ays/common/util/AysDataProvider.java [hunk 2]
```diff
@@ -237,6 +237,16 @@ public Object[][] invalidEmailAddress() {
         };
     }
 
+    @org.testng.annotations.DataProvider(name = "invalidPasswordAndEdgeCases")
```
Comment (Nbhtozcn) [inline src/test/java/org/ays/common/util/AysDataProvider.java:240]: @org.testng.annotations.DataProvider(name = "invalidPasswordData")

File: src/test/java/org/ays/common/util/AysRandomUtil.java [hunk 3]
```diff
@@ -29,6 +29,17 @@ public static String generatePassword() {
...
+    public static String generatePasswordWithinRange(int minLength, int maxLength) {
+        if (minLength < 1 || maxLength < minLength) {
+            throw new IllegalArgumentException("Invalid password length range." +
+                    " Ensure minLength >= 1 and maxLength >= minLength.");
+        }
+        if (minLength == maxLength) {
+            return FAKER.internet().password(minLength, minLength + 1).substring(0, minLength);
+        }
+        return FAKER.internet().password(minLength, maxLength);
+    }
```
Comment (agitrubard) [inline src/test/java/org/ays/common/util/AysRandomUtil.java:41]: Okunurluluğu arttırmak için boşluklara dikkat edebiliriz. Ek olarak metot adına `generatePassword` denebilir. Java method overloading'e izin verdiği için aynı isimde kullanılabilir. Parametre almadığı zaman parametresiz metot çalışacaktır.
```suggestion
    public static String generatePassword(int minLength, int maxLength) {
    
        if (minLength < 1 || maxLength < minLength) {
            throw new IllegalArgumentException("Invalid password length range." +
                    " Ensure minLength >= 1 and maxLength >= minLength.");
        }
        
        if (minLength == maxLength) {
            return FAKER.internet().password(minLength, minLength + 1).substring(0, minLength);
        }
        
        return FAKER.internet().password(minLength, maxLength);
    }
```

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/test/java/org/ays/common/util/AysRandomUtil.java (hunk 3 above)
Comment (htczrrsz) [reply src/test/java/org/ays/common/util/AysRandomUtil.java:41]: @agitrubard overloading bilerek yapmamistim aslinda, metoda deger girdiğimizde o degerlerin ne icin oldukları daha anlasilir olsun diye. Ama düzeltebilirim gerek yok diyorsan ?

**Review by Nbhtozcn — APPROVED** [review]

File: src/test/java/org/ays/common/util/AysRandomUtil.java (hunk 3 above)
Comment (agitrubard) [reply src/test/java/org/ays/common/util/AysRandomUtil.java]: @htczrrsz Değerlerin isminden ne için oldukları anlaşılıyor aslında o yüzden özel olarak metot adında belirtmemize ihtiyaç yok ✅

**Review by agitrubard — APPROVED** [review]
